content/intro-to-storybook/ember/en/screen.md

```diff
@@ -63,78 +76,106 @@ export const reducer = (state, action) => {
     case actions.SET_ERROR:
       return {
         ...state,
-        isError: true,
-      };
+        isError: true
+      }
+    case actions.SET_LOADING:
+      return {
+        ...state,
+        isLoading: true
+      }
     default:
       return state || initialState;
   }
 };
 
-export default combineReducers({
-  reducer,
-});
+export default reducers;
 ```
 
-The store is updated with the new field. Let's create a presentational `pure-inbox-screen.hbs` in our `app/components` folder:
+As our store is updated with the new fields let's move on creating an
+`inbox-screen.hbs` component in our `app/components` folder:
 
 ```handlebars
-{{!--app/components/pure-inbox-screen.hbs--}}
-<div class="page lists-show">
-  <nav>
-    <h1 class="title-page">
-      <span class="title-wrapper">
-        Taskbox
-      </span>
-    </h1>
-  </nav>
-  {{#if @error}}
-    <div class="page lists-show">
-      <div class="wrapper-message">
-        <span class="icon-face-sad"></span>
-        <div class="title-message">
-          Oh no!
-        </div>
-        <div class="subtitle-message">
-          Something went wrong
+{{!--app/components/inbox-screen.hbs--}}
+<div>
+  <div class="page lists-show">
+    <nav>
+      <h1 class="title-page">
+        <span class="title-wrapper">
+          Taskbox
+        </span>
+      </h1>
+    </nav>
+    {{#if this.loading}}
+      <LoadingRow />
+      <LoadingRow />
+      <LoadingRow />
+      <LoadingRow />
+      <LoadingRow />
+    {{else if this.error}}
+      <div class="page lists-show">
+        <div class="wrapper-message">
+          <span class="icon-face-sad"></span>
+          <div class="title-message">
+            Oh no!
+          </div>
+          <div class="subtitle-message">
+            Something went wrong
+          </div>
         </div>
       </div>
-    </div>
-  {{else}}
-    <TaskList />
-  {{/if}}
+    {{else}}
+      <TaskList
+        @tasks={{this.tasks}}
+        @pinTask={{this.pinTask}}
+        @archiveTask={{this.archiveTask}}
+      />
+    {{/if}}
+  </div>
 </div>
 ```
 
-Then, we can create a container, which again grabs the data for the `PureInboxScreen` in `app/components/inbox-screen.hbs` add the following:
-
-```handlebars
-{{!--app/components/inbox-screen.hbs --}}
-<div>
-  <PureInboxScreen @error={{this.error}} />
-</div>
-```
+It already has sections to show various states for loading and errors. Next up
```

Comment:
@gossi probably phrase this a bit different. Something like: "We now have a way to handle the various states of our application. We can now move the existing logic for handling the data loading and error handling into the component." What do you think?